app/views/MainTemplate.scala

```diff
@@ -0,0 +1,133 @@
+import models.ActingAsAttorneyFor
+import models.admin.SCAWrapperToggle
+import play.api.Logging
+import play.api.i18n.Messages
+import play.api.mvc.Request
+import play.twirl.api.{Html, HtmlFormat}
+import uk.gov.hmrc.mongoFeatureToggles.services.FeatureFlagService
+import uk.gov.hmrc.play.http.HeaderCarrierConverter
+import uk.gov.hmrc.sca.models.BannerConfig
+import uk.gov.hmrc.sca.services.WrapperService
+import views.html.components.HeadBlock
+import views.html.includes.sidebar
+import views.html.nonScaWrapperMain
+
+import javax.inject.Inject
+import scala.concurrent.Await
+import scala.concurrent.duration.{Duration, SECONDS}
+
+@ImplementedBy(classOf[MainTemplateImpl])
+trait MainTemplate {
+
+  def apply(
+    pageTitle: String,
+    backLinkHref: Option[String] = None,
+    disableSessionExpired: Boolean = false,
+    additionalScripts: Option[Html] = None,
+    backLinkAttrs: Map[String, String] = Map.empty,
+    actingAttorney: Option[ActingAsAttorneyFor] = None,
+    beforeContentHtml: Option[Html] = None
+  )(contentBlock: Html)(implicit
+    request: Request[_],
+    messages: Messages
+  ): HtmlFormat.Appendable
+}
+
+class MainTemplateImpl @Inject() (
+  appConfig: ApplicationConfig,
+  featureFlagService: FeatureFlagService,
+  wrapperService: WrapperService,
+  oldLayout: nonScaWrapperMain,
+  headBlock: HeadBlock,
+  sidebar: sidebar
+) extends MainTemplate
+    with Logging {
+  override def apply(
+    pageTitle: String,
+    backLinkHref: Option[String] = None,
+    disableSessionExpired: Boolean = false,
+    additionalScripts: Option[Html] = None,
+    backLinkAttrs: Map[String, String] = Map.empty,
+    actingAttorney: Option[ActingAsAttorneyFor] = None,
+    beforeContentHtml: Option[Html] = None
+  )(contentBlock: Html)(implicit request: Request[_], messages: Messages): HtmlFormat.Appendable = {
+    val scaWrapperToggle =
+      Await.result(featureFlagService.get(SCAWrapperToggle), Duration(appConfig.SCAWrapperFutureTimeout, SECONDS))
+    val fullPageTitle    = s"$pageTitle - ${Messages("generic.ats.browser.title")}"
+
+    if (scaWrapperToggle.isEnabled) {
+      logger.debug(s"SCA Wrapper layout used for request `${request.uri}``")
+
+      val showAccountMenu = actingAttorney.isEmpty && !disableSessionExpired
+
+      val (signOutUrlString: String, showSignOut: Boolean) = request match {
+        case _: AuthenticatedRequest[_]     =>
+          (controllers.routes.AccountController.signOut.url, true)
+        case _: PayeAuthenticatedRequest[_] =>
+          (controllers.routes.AccountController.signOut.url, true)
+        case _                              => ("", false)
+      }
+
+      wrapperService.layout(
+        content = contentBlock,
+        pageTitle = Some(fullPageTitle),
+        serviceNameKey = Some(messages("generic.ats.browser.title")),
+        serviceNameUrl = Some(appConfig.serviceUrl),
+        sidebarContent = Some(sidebar(beforeContentHtml)),
```

Comment:
This isn't a full sidebar thing, it's supposed to be before the main content, may need to be some weird workaround for this
<img width="1125" alt="Screenshot 2023-08-23 at 09 02 32" src="https://github.com/hmrc/tax-summaries-frontend/assets/47452401/808a1701-9fce-49a7-8f3e-a10572c78fc2">